src/main/java/mate/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+package mate.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.lib.Dao;
+import mate.model.Manufacturer;
+import mate.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturerRequest = "INSERT INTO manufacture(name, country) values(?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufactureStatement =
+                        connection.prepareStatement(insertManufacturerRequest,
+                                Statement.RETURN_GENERATED_KEYS);) {
+
+            createManufactureStatement.setString(1, manufacturer.getName());
+            createManufactureStatement.setString(2, manufacturer.getCountry());
+            createManufactureStatement.executeUpdate();
+            ResultSet generatedKeys = createManufactureStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+
+        } catch (SQLException e) {
+            throw new RuntimeException("Can`t insert manufacture to db", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String sqlSelect = "SELECT * FROM manufacture WHERE id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement =
+                        connection.prepareStatement(sqlSelect);) {
+
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement
+                    .executeQuery();
+            Manufacturer manufacturer = new Manufacturer();
+            if (resultSet.next()) {
+                String name = resultSet.getString("name");
+                String country = resultSet.getString("country");
+                Long resultId = resultSet.getObject("id", Long.class);
+                manufacturer.setName(name);
+                manufacturer.setCountry(country);
+                manufacturer.setId(resultId);
+                return Optional.of(manufacturer);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new RuntimeException("Can`t get all manufactures from DB", e);
```

Comment:
all? that`s not getAll method
```suggestion
            throw new RuntimeException("Can`t get manufacture from DB by id - " + id, e);
```